server/base/src/main/java/org/apache/accumulo/server/ServerContext.java

```diff
@@ -112,7 +125,26 @@ public synchronized ServerConfigurationFactory getServerConfFactory() {
 
   @Override
   public AccumuloConfiguration getConfiguration() {
-    return getServerConfFactory().getSystemConfiguration();
+    if (systemConfig == null) {
+      ZooCache propCache = new ZooCache(getZooKeepers(), getZooKeepersSessionTimeOut());
+      systemConfig = new ZooConfiguration(this, propCache, getSiteConfiguration());
+    }
+    return systemConfig;
+  }
+
+  public TableConfiguration getTableConfiguration(TableId id) {
+    return getServerConfFactory().getTableConfiguration(id);
+  }
+
+  public NamespaceConfiguration getNamespaceConfiguration(NamespaceId namespaceId) {
+    return getServerConfFactory().getNamespaceConfiguration(namespaceId);
+  }
+
+  public DefaultConfiguration getDefaultConfiguration() {
+    if (defaultConfig == null) {
+      defaultConfig = DefaultConfiguration.getInstance();
+    }
+    return defaultConfig;
```

Comment:
I don't think there's any reason to load this singleton lazily.

```suggestion
    return DefaultConfiguration.getInstance();
```

Reply:
Perhaps this [method](https://github.com/apache/accumulo/blob/3fd5cad92f9b63ac19e4466f3f2d5237b905262c/core/src/main/java/org/apache/accumulo/core/conf/DefaultConfiguration.java#L43) is poorly named because it just returns a new object:
```
public static DefaultConfiguration getInstance() {
    return new DefaultConfiguration();
  }
```
I think it is worth storing a local copy to prevent creating too many identical objects.

Reply:
The class is documented as a singleton... but that appears to have stopped being the case as of 37f900b7550d6ffe7b39f2ff5cd766912ab09141 ; it should be a singleton. It doesn't need to do all the synchronization that the commit cleaned up, but it should still save and reuse the single instance.

Reply:
Yeah and since it anything loading the class will use it, I think it could just be created once during class load as a static variable.